pkg/operator/reports_test.go

```diff
@@ -92,3 +99,224 @@ func TestGetNextReportPeriod(t *testing.T) {
+			},
+			LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 2, 0)},
+		}, schedule, false), expectFlag: true},
+	}
+
+	for _, testCase := range testTable {
+		var mockLogger = logrus.New()
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			runningCond := isReportFinished(mockLogger, testCase.report)
+			assert.Equalf(t, runningCond, testCase.expectFlag, "expected the report '%s' would return '%t', but got '%t'", testCase.name, testCase.expectFlag, runningCond)
+		})
+	}
+}
+
+func TestValidateReport(t *testing.T) {
+	const (
+		testNamespace           = "default"
+		testReportName          = "test-report"
+		testQueryName           = "test-query"
+		testGenerationQueryName = "test-GQN"
+	)
+
+	reportStart := &time.Time{}
+	reportEndTmp := reportStart.AddDate(0, 1, 0)
+	reportEnd := &reportEndTmp
+
+	testTable := []struct {
+		name         string
+		report       *metering.Report
+		expectFlag   bool
+		expectErrMsg string
+	}{
+		{name: "empty spec.generationQuery returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, "", reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectFlag:   true,
+			expectErrMsg: "must set spec.generationQuery",
+		},
+		{name: "spec.ReportingStart > spec.ReportingEnd returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportEnd, reportStart, v1alpha1.ReportStatus{}, nil, false),
+			expectFlag:   true,
+			expectErrMsg: fmt.Sprintf("spec.reportingEnd (%s) must be after spec.reportingStart (%s)", reportStart.String(), reportEnd.String()),
+		},
+		{name: "spec.ReportingEnd is unset and spec.RunImmediately is set returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, nil, v1alpha1.ReportStatus{}, nil, true),
+			expectFlag:   true,
+			expectErrMsg: "spec.reportingEnd must be set if report.spec.runImmediately is true",
+		},
+		{name: "spec.GenerationQueryName does not exist returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectFlag:   true,
+			expectErrMsg: fmt.Sprintf("ReportGenerationQuery %s does not exist", testGenerationQueryName),
+		},
+		{name: "invalid ReportGenerationQuery dependencies returns err",
+			report:       testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{}, nil, false),
+			expectFlag:   true,
+			expectErrMsg: fmt.Sprintf("failed to validate ReportGenerationQuery dependencies %s: %v", testGenerationQueryName, "stub"),
+		},
+	}
+
+	// hardcore the report spec's generation query name for tests that need that
+	testTable[3].report.Spec.GenerationQueryName = testGenerationQueryName
+	testTable[4].report.Spec.GenerationQueryName = testGenerationQueryName
+
+	ds1 := testhelpers.NewReportDataSource("datasource1", testNamespace)
```

Comment:
We already test the validate dependencies code on it's own thoroughly, so you don't need to re-test that here, so you can keep the resources used for testing fairly minimal .